### PR TITLE
add constant_value_units for prob. forecasts

### LINKED ISSUE #470: add attribute constant_value_units to prob fx
`constant_value_units` would be set dynamically based on the `axis` and `variable` values. Also `units` should be set to `%` if `axis='x'`.

related to https://github.com/SolarArbiter/solarforecastarbiter-api/issues/251

### PR DESCRIPTION
**Pull request by alorenzo175**
- [x] Closes #470 .
  - [x] I am familiar with the [contributing guidelines](https://solarforecastarbiter-core.readthedocs.io/en/latest/contributing.html).
  - [x] Tests added.
  - [x] Updates entries to [`docs/source/api.rst`](https://github.com/SolarArbiter/solarforecastarbiter-core/blob/master/docs/source/api.rst) for API changes.
  - [x] Adds descriptions to appropriate "what's new" file in [`docs/source/whatsnew`](https://github.com/SolarArbiter/solarforecastarbiter-core/tree/master/docs/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
  - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
  - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

### CODE REVIEW

File: solarforecastarbiter/datamodel.py [hunk 1]
```diff
@@ -840,10 +842,19 @@ def __post_init__(self):
         super().__post_init__()
 
 
+def __set_constant_value_units__(cls):
+    if cls.axis == 'x':
+        object.__setattr__(cls, 'constant_value_units', cls.units)
```
Comment (wholmgren) [inline solarforecastarbiter/datamodel.py:847]:
```suggestion
        # e.g. Prob(o < 10 MW). Forecast is in %, constant value is 10 MW
        object.__setattr__(cls, 'constant_value_units', cls.units)
```

File: solarforecastarbiter/datamodel.py [hunk 2]
```diff
@@ -840,10 +842,19 @@ def __post_init__(self):
...
+        object.__setattr__(cls, 'units', '%')
+    else:
+        object.__setattr__(cls, 'constant_value_units', '%')
```
Comment (wholmgren) [inline solarforecastarbiter/datamodel.py:850]:
```suggestion
        # e.g. Prob(o < f) = 90%. Forecast is in units of obs, constant value is %
        object.__setattr__(cls, 'constant_value_units', '%')
```

File: solarforecastarbiter/datamodel.py [hunk 3]
```diff
@@ -1239,8 +1253,13 @@ def __check_axis_consistency__(axis, constant_values):
...
         return
-    ref_unit = args[0].units
-    if not all(arg.units == ref_unit for arg in args):
+    unique_units = set()
+    for arg in args:
+        if getattr(arg, 'axis', None) == 'x':
+            unique_units.add(arg.constant_value_units)
+        else:
+            unique_units.add(arg.units)
+    if not len(unique_units) == 1:
```
Comment (wholmgren) [inline solarforecastarbiter/datamodel.py:1262]: is this style preferable to `!= 1`?

**Review by wholmgren — APPROVED** [review]

File: solarforecastarbiter/datamodel.py (hunk 3 above)
Comment (alorenzo175) [reply solarforecastarbiter/datamodel.py:1262]: nope

**Pull request merged**